### ISSUE TITLE
Python3.6 issue when pickling vm data

### ISSUE THREAD

**Issue by beraldoleal** [avocado-framework/avocado-vt#3256]
**Describe the bug**

tl;dr: Avocado-vt is pickling vm data, however pickle library when using python3.6 is not able to pickle _thread.RLock objects.

`virttest/qemu_vm.py:3093` (and probably other places) is saving into the `self.process` the return of an aexpect execution (more specifically, a `Tail` object). Along with this tail object we used to have a logging function (i.e `<function logging.info(...)>`). So in the context of a vm we had: `vm.process.output_func = <function logging.info(...)>`, which could be "pickled".

After adbfa568683c00205cf1c1ca83b19c29e527ddef, we are saving a "bound method", so `vm.process.output_func = <bound method Logger.info of <Logger avocado.virttest.qemu_VM (DEBUG)>>`, which is attached to the specific logger object, and because of a Python3.6 `threading.RLock` issue, cannot be "pickled".

**Steps to reproduce**

Using Python3.6, just run:

```bash
$ avocado run boot
```

If you would like to isolate the problem, here is another reproducible without Avocado or Avocado-vt:

```python
import pickle
import logging
 
if __name__ == '__main__':
    log = logging.getLogger()
    log.addHandler(logging.FileHandler("/tmp/test.data", mode="w"))

    with open('/tmp/foo', 'wb') as fp: 
        pickle.dump(log, fp, protocol=0)
```

**Expected behavior**

```
JOB ID     : ea6bed4d658039e2eea31fe599d232e6faac6963
JOB LOG    : /home/local/avocado/job-results/job-2021-10-15T14.46-ea6bed4/job.log
 (1/1) io-github-autotest-qemu.boot: STARTED
 (1/1) io-github-autotest-qemu.boot: PASS
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/local/avocado/job-results/job-2021-10-15T14.46-ea6bed4/results.html
JOB TIME   : 21.13 s
```

**Current behavior**
```
JOB ID     : ea6bed4d658039e2eea31fe599d232e6faac6963
JOB LOG    : /home/local/avocado/job-results/job-2021-10-15T14.46-ea6bed4/job.log
 (1/1) io-github-autotest-qemu.boot: STARTED
 (1/1) io-github-autotest-qemu.boot: ERROR: can't pickle _thread.RLock objects (19.66 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/local/avocado/job-results/job-2021-10-15T14.46-ea6bed4/results.html
JOB TIME   : 21.13 s
```

**System information (please complete the following information):**
 - OS: `lsb_release -a`: Fedora 33
 - Avocado version: `avocado -v`: avocado 91
 - Avocado installation method: pip, rpm, github ?: python3 setup.py develop
 
**Additional information**

We need to double check if python3.6 is really needed at this point.
